pkg/assembler/backends/ent/backend/neighbors.go

```diff
@@ -28,14 +33,91 @@ func (b *EntBackend) Node(ctx context.Context, node string) (model.Node, error)
 	case *ent.Artifact:
 		return toModelArtifact(v), nil
 	case *ent.PackageVersion:
-		return toModelPackage(backReferencePackageVersion(v)), nil
+		pv, err := b.client.PackageVersion.Query().
+			Where(packageversion.ID(v.ID)).
+			WithName(func(q *ent.PackageNameQuery) {
+				q.WithNamespace(func(q *ent.PackageNamespaceQuery) {
+					q.WithPackage()
+				})
+			}).
+			Only(ctx)
+		if err != nil {
+			return nil, err
+		}
+		return toModelPackage(backReferencePackageVersion(pv)), nil
 	case *ent.PackageName:
-		return toModelPackage(backReferencePackageName(v)), nil
+		pn, err := b.client.PackageName.Query().
+			Where(packagename.ID(v.ID)).
+			WithNamespace(func(q *ent.PackageNamespaceQuery) {
+				q.WithPackage()
+			}).
+			WithVersions().
+			Only(ctx)
+		if err != nil {
+			return nil, err
+		}
+		return toModelPackage(backReferencePackageName(pn)), nil
+	case *ent.PackageNamespace:
+		pns, err := b.client.PackageNamespace.Query().
+			Where(packagenamespace.ID(v.ID)).
+			WithPackage().
+			WithNames(func(q *ent.PackageNameQuery) {
+				q.WithVersions()
+			}).
+			Only(ctx)
+		if err != nil {
+			return nil, err
+		}
+		return toModelPackage(backReferencePackageNamespace(pns)), nil
 	case *ent.PackageType:
-		return toModelPackage(v), nil
+		pt, err := b.client.PackageType.Query().
+			Where(packagetype.ID(v.ID)).
+			WithNamespaces(func(q *ent.PackageNamespaceQuery) {
+				q.WithNames(func(q *ent.PackageNameQuery) {
+					q.WithVersions()
```

Comment:
I think we need to constrain this somehow... otherwise it could potentially query a lot of stuff. Is node by PackageType a realistic query to support?

Reply:
As per the discussion on Slack, this has been refactored to just return the `package namespace` and nothing "deeper" than this.